diff -Naur Catalyst-Authentication-Credential-OpenID-0.11/lib/Catalyst/Authentication/Credential/OpenID.pm Catalyst-Authentication-Credential-OpenID-new/lib/Catalyst/Authentication/Credential/OpenID.pm
--- Catalyst-Authentication-Credential-OpenID-0.11/lib/Catalyst/Authentication/Credential/OpenID.pm	2008-10-05 09:44:22.000000000 +0200
+++ Catalyst-Authentication-Credential-OpenID-new/lib/Catalyst/Authentication/Credential/OpenID.pm	2008-11-30 23:43:16.000000000 +0100
@@ -8,7 +8,7 @@
     __PACKAGE__->mk_accessors(qw/ _config realm debug secret /);
 }
 
-our $VERSION = "0.11";
+our $VERSION = "0.12";
 
 use Net::OpenID::Consumer;
 use Catalyst::Exception ();
@@ -72,6 +72,9 @@
         my $identity = $csr->claimed_identity($claimed_uri)
             or Catalyst::Exception->throw($csr->err);
 
+        $identity->set_extension_args(%{$self->_config->{extensions}})
+            if(defined %{$self->_config->{extensions}});
+
         my $check_url = $identity->check_url(
             return_to  => $current . '?openid-check=1',
             trust_root => $current,
@@ -96,6 +99,10 @@
             # This is where we ought to build an OpenID user and verify against the spec.
             my $user = +{ map { $_ => scalar $identity->$_ }
                 qw( url display rss atom foaf declared_rss declared_atom declared_foaf foafmaker ) };
+            
+            for(keys %{$self->{_config}->{extensions}}) {
+                $user->{extensions}->{$_} = $identity->signed_extension_fields($_);
+            }
 
             my $user_obj = $realm->find_user($user, $c);
 
diff -Naur Catalyst-Authentication-Credential-OpenID-0.11/t/TestApp/lib/TestApp.pm Catalyst-Authentication-Credential-OpenID-new/t/TestApp/lib/TestApp.pm
--- Catalyst-Authentication-Credential-OpenID-0.11/t/TestApp/lib/TestApp.pm	2008-10-05 09:40:53.000000000 +0200
+++ Catalyst-Authentication-Credential-OpenID-new/t/TestApp/lib/TestApp.pm	2008-11-30 23:18:50.000000000 +0100
@@ -47,6 +47,13 @@
                       whitelisted_hosts => [qw/ 127.0.0.1 localhost /],
                       timeout => 10,
                   },
+                  extensions => {
+                     'http://openid.net/extensions/sreg/1.1',
+                     {
+                        required => 'email',
+                        optional => 'fullname,nickname,timezone',
+                     }
+                  },
                   debug => 1,
                   credential => {
                       class => "OpenID",
